modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/processor/OperationMessageProcessor.java

```diff
@@ -167,4 +183,62 @@ public void setMuleContext(MuleContext muleContext)
         this.muleContext = muleContext;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Result<List<MetadataKey>> getMetadataKeys() throws MetadataResolvingException
+    {
+        return metadataMediator.getMetadataKeys(getMetadataContext());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Result<MetadataType> getContentMetadata(MetadataKey key) throws MetadataResolvingException
+    {
+        return metadataMediator.getContentMetadata(getMetadataContext(), key);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Result<MetadataType> getOutputMetadata(MetadataKey key) throws MetadataResolvingException
+    {
+        return metadataMediator.getOutputMetadata(getMetadataContext(), key);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Result<OperationMetadataDescriptor> getMetadata() throws MetadataResolvingException
+    {
+        return metadataMediator.getMetadata();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Result<OperationMetadataDescriptor> getMetadata(MetadataKey key) throws MetadataResolvingException
+    {
+        return metadataMediator.getMetadata(getMetadataContext(), key);
+    }
+
+    private MetadataContext getMetadataContext() throws MetadataResolvingException
+    {
+        //FIXME bug when the configuration is accesed by the extensionModel, but the implicit config is dynamic
+        if (!StringUtils.isBlank(configurationProviderName) &&
+            muleContext.getRegistry().get(configurationProviderName) instanceof DynamicConfigurationProvider)
+        {
+            throw new MetadataResolvingException("Configuration used for Metadata fetch cannot be dynamic", FailureType.INVALID_CONFIGURATION);
```

Comment:
@marianogonzalez any ideas on how to cover the case where providerName is blank and extensionModel is used instead to resolve the Config without duplicating all [this logic](https://github.com/mulesoft/mule/blob/sdk-support-for-datasense/modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java#L174)?